llvm/lib/Target/X86/X86RegisterInfo.cpp

```diff
@@ -1082,6 +1088,42 @@ bool X86RegisterInfo::getRegAllocationHints(Register VirtReg,
       VirtReg, Order, Hints, MF, VRM, Matrix);
 
   unsigned ID = RC.getID();
+
+  if (!VRM || DisableRegAllocHints)
```

Comment:
Hmm, do not disable hints for other instructions such as AMX in this patch.

Reply:
c788607